application/cms/scanner_service.py

```diff
@@ -0,0 +1,55 @@
+import json
+
+import requests
+
+from application.cms.exceptions import (
+    UnknownFileScanStatus,
+    UploadCheckPending,
+    UploadCheckFailed,
+    UploadCheckVirusFound,
+)
+from application.cms.service import Service
+
+
+class ScannerService(Service):
+    def init_app(self, app):
+        super().init_app(app)
+        self.base_url = self.app.config["ATTACHMENT_SCANNER_URL"]
+        self.token = self.app.config["ATTACHMENT_SCANNER_API_KEY"]
+        self.enabled = self.app.config["ATTACHMENT_SCANNER_ENABLED"]
+
+    def scan_file(self, filename, fileobj) -> bool:
+        """
+        return: True if scanned and safe; False if not scanned
+        raises: child of UploadException if scanned and a problem occurred
+        """
+        if self.enabled:
+            response = requests.post(
+                f"{self.base_url}/requests", headers={"Authorization": f"Bearer {self.token}"}, files={"file": fileobj}
+            )
+
+            status = response.json()["status"].lower()
+
+            if status == "ok":
```

Comment:
These status messages could be factored out to class-level constants maybe, but I guess they shouldn't change so :man_shrugging: